Kluwer Arbitration.js

```diff
@@ -0,0 +1,1011 @@
+async function getDocumentDetails(documentID) {
+	let detailsURL = "https://www.kluwerarbitration.com/api/Document?id=" + documentID;
+	Z.debug("Requesting document details");
+	Z.debug(detailsURL);
+	return ZU.requestJSON(detailsURL);
+}
+
+// Get the JSON object describing the standalone case (i.e., not contained in a journal)
+async function getCaseDetails(caseID) {
+	let detailsURL = "https://www.kluwerarbitration.com/case/GetDetailsByCaseId?caseId=" + caseID;
+	Z.debug("Requesting case details");
+	Z.debug(detailsURL);
+	return ZU.requestJSON(detailsURL);
+}
+
+// Get the JSON object of the case ToC (history)
+async function getCaseToc(caseID) {
+	let detailsURL = "https://www.kluwerarbitration.com/case/" + caseID + "/GetToc";
+	Z.debug("Requesting case ToC");
+	Z.debug(detailsURL);
+	return ZU.requestJSON(detailsURL);
+}
+
+async function getProceedingDetails(proceedingID) {
+	let detailsURL = "https://www.kluwerarbitration.com/proceeding/" + proceedingID;
+	Z.debug("Requesting proceeding details");
+	Z.debug(detailsURL);
+	return ZU.requestJSON(detailsURL);
+}
+
+function getDocId(url) {
+	const segments = new URL(url).pathname.split('/');
+	const documentID = segments.pop() || segments.pop(); // Handle potential trailing slash
+	Z.debug("Document ID: " + documentID);
+	return documentID;
+}
+
+function getCaseId(url) {
+	const path = new URL(url).pathname;
+	const regex = /case\/(\d+)/;
+	let caseID
+	if ((caseID = regex.exec(path)) !== null) {
+		Z.debug("Case ID: " + caseID[1]);
+		return caseID[1];
+	}
+	return null;
+}
+
+function getProceedingId(url) {
+	const path = new URL(url).pathname;
+	const regex = /case\/(\d+\/\d+)/;
+	let proceedingID
+	if ((proceedingID = regex.exec(path)) !== null) {
+		Z.debug("Proceeding ID: " + proceedingID[1]);
+		return proceedingID[1];
+	}
+	return null;
+}
+
+async function getTypeAndDetails(doc, url) {
+	if (url.includes('document/case')) {
+		Z.debug('Found a case');
+
+		/*let caseID = getCaseId(url);
+		let documentID = getDocId(url);
+		let caseDetails = await getCaseDetails(caseID);
+		//let caseToc = await getCaseToc(caseID);
+		let docDetails = await getDocumentDetails(documentID);*/
+		return { type: 'case', pubDetails: null, docDetails: null };
+	}
+	else if (url.includes('/document/')) {
+		Z.debug('Found a document');
+		
+		let documentID = getDocId(url);
+		let pubDetails = await getPublicationDetails(documentID);
+		let docDetails = await getDocumentDetails(documentID);
+		
+		//let pubType = pubDetails.publicationType;
+		let zotType = getType(pubDetails, docDetails);
+		if (zotType) {
+			Z.debug("Document is: " + zotType);
+			return { type: zotType, pubDetails: pubDetails, docDetails: docDetails };
+		}
+
+		/*else {
+			Z.debug(pubType + " not yet suppported");
+		}*/
+	}
+	else if (url.includes('/search')) {
+		Z.debug("Investigating search page");
+		let selector = 'div#vueApp';
+		Z.debug("Monitoring " + selector);
+		Z.monitorDOMChanges(doc.querySelector(selector));
```

Comment:
The whole visible website is actually contained in that `div` so AFAIK, it's always there. I used it to avoid relying on class names or `body` (even though this `div` is barely more specific than `body`).